src/Services/LoggerFactory.php

```diff
@@ -79,7 +91,8 @@ public function __construct(SettingGateway $settingGateway)
     {
         $this->settingGateway = $settingGateway;
         $this->filePath = $settingGateway->getSettingByScope('System', 'absolutePath') . DIRECTORY_SEPARATOR . 'var' . DIRECTORY_SEPARATOR . 'log' . DIRECTORY_SEPARATOR;
-        $this->loggerLevel = $settingGateway->getSettingByScope('System', 'installType') === 'Production' ? Logger::WARNING : Logger::DEBUG;
+        $this->installType = $settingGateway->getSettingByScope('System', 'installType');
```

Comment:
I wonder if the SettingGateway dependency used in this class creates a tight coupling between it and the LoggerFactory. Another approach could be to use the DI container to pass in/set only the necessary configuration when the factory is constructed.